hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/BaseLoadBalancer.java

```diff
@@ -1223,6 +1227,20 @@ protected boolean areSomeRegionReplicasColocated(Cluster c) {
     return false;
   }
 
+  protected final synchronized boolean idleRegionExist(Cluster c){
```

Comment:
why is this synchronized?

Reply:
Hmm, as part of HBASE-12559, many methods that are overridden in `StochasticLoadBalancer` are made `synchronized` due to findbugs warnings. I was surprised to see all these methods non-synchronized in parent and synchronized in child class.

Reply:
Yes, I've got a bit confused about this, and as I've originally put this in `StochasticLoadBalancer`, I thought it it would work there as well. 
Should I remove the `synchronised`?

Reply:
You have already removed `synchronized` and it's fine to do so, anyways method is now in BaseLoadBalancer.